### PR TITLE
Fix OpenSSH FP.

### PR DESCRIPTION
**Pull request by nmasdoufi-ol**

### CODE REVIEW

File: agent/exploits/cve_2024_7589.py
Comment (rabsondev) [inline agent/exploits/cve_2024_7589.py:1]: Where's the unit test for this change?

File: agent/exploits/cve_2024_7589.py [hunk 1]
```diff
@@ -75,22 +74,13 @@ def _create_vulnerability(
...
-    match = re.search(r"OpenSSH_(\d+\.\d+(p\d+)?)", version_string)
-    return match.group(1) if match is not None else ""
-
-
 def _is_vulnerable_version(version_string: str) -> bool:
     """Checks if the version is vulnerable."""
-    version = _parse_version(version_string)
-    if version is not None:
-        if version in VULNERABLE_VERSIONS:
+    if version_string is not None:
```
Comment (deadly-panda) [inline agent/exploits/cve_2024_7589.py:79]: I would rename the arg to `ssh_banner` or `banner` ..

Comment (nmasdoufi-ol) [reply agent/exploits/cve_2024_7589.py:79]: It's used elsewhere I didn't want any confuse.

File: tests/exploits/cve_2024_7589_test.py [hunk 2]
```diff
@@ -20,9 +20,7 @@ def testAcceptExploit_whenVulnerableBanner_shouldReturnTrue(
 
     accept = exploit_instance.accept(target)
 
-    assert (
-        accept is True
-    ), f"Expected vulnerability detection for IP address {target.origin}, but accept returned False"
+    assert accept is True
```
Comment (3asm) [inline tests/exploits/cve_2024_7589_test.py:23]: How come the unit test don't reflect the change?

**Review by 3asm — CHANGES_REQUESTED** [review]

File: tests/exploits/cve_2024_7589_test.py (hunk 2 above)
Comment (nmasdoufi-ol) [reply tests/exploits/cve_2024_7589_test.py:23]: The problem was in the logic of `_is_vulnerable_version` method. The unit tests take a vulnerable version and a non vulnerable version. I don't understand what change that must be reflected in the unit tests?

File: agent/exploits/cve_2024_7589.py [hunk 3]
```diff
@@ -106,7 +98,7 @@ def accept(self, target: definitions.Target) -> bool:
         logger.debug("Retrieved banner for %s:%d - %s", ip_address, port, banner)
 
         if "OpenSSH" in banner:
-            if _is_vulnerable_version(banner) is True:
+            if _is_vulnerable_version(banner):
```
Comment (ybadaoui-ostorlab) [inline agent/exploits/cve_2024_7589.py:101]: explicit!

**Pull request merged**